Data/Functor/Foldable.hs

```diff
@@ -448,12 +449,18 @@ grefold w m f g a = ghylo w m f g a
 futu :: Corecursive t => (a -> Base t (Free (Base t) a)) -> a -> t
 futu = gana distFutu
 
+gfutu :: (Corecursive t, Monad m) => (forall b. m (Base t b) -> Base t (m b)) -> (a -> Base t (FreeT (Base t) m a)) -> a -> t
+gfutu g = gana (distGFutu g)
+
 distFutu :: Functor f => Free f (f a) -> f (Free f a)
-distFutu = distGFutu id
+distFutu (Pure fx) = Pure <$> fx
+distFutu (Free ff) = Free . distFutu <$> ff
```

Comment:
`distGFutu distAna` would be a valid implementation if `Free f` was defined as `FreeT f Identity`